interface/ceed-preconditioning.c

```diff
@@ -1073,6 +1074,47 @@ int CeedOperatorLinearAssembleQFunction(CeedOperator op, CeedVector *assembled,
   }
 }
 
+/**
+  @brief Update vector containing previously assembled linear CeedQFunction
+           associated with a CeedOperator
+
+  Note: Calling this function asserts that setup is complete
+          and sets the CeedOperator as immutable.
+
+  @param op              CeedOperator to assemble CeedQFunction
+  @param assembled       CeedVector to store assembled CeedQFunction at
+                           quadrature points
+  @param rstr            CeedElemRestriction for CeedVector containing assembled
+                           CeedQFunction
+  @param request         Address of CeedRequest for non-blocking completion, else
+                           @ref CEED_REQUEST_IMMEDIATE
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref User
+**/
+int CeedOperatorLinearAssembleQFunctionUpdate(CeedOperator op,
+    CeedVector assembled, CeedElemRestriction rstr, CeedRequest *request) {
+  int ierr;
+  ierr = CeedOperatorCheckReady(op); CeedChk(ierr);
+
+  // Backend version
+  if (op->LinearAssembleQFunction) {
+    ierr = op->LinearAssembleQFunction(op, true, &assembled, &rstr, request);
+    CeedChk(ierr);
+    return CEED_ERROR_SUCCESS;
```

Comment:
Am I missing something big or do you want this to be `false` so it assembles with the caller's provided vector and restriction?
```suggestion
    ierr = op->LinearAssembleQFunction(op, false, &assembled, &rstr, request);
    CeedChk(ierr);
    return CEED_ERROR_SUCCESS;
```

Reply:
Yeah, that will be switched over to false. I'll be expanding this commit to hold the vector and elem restriction for operator diagonal assembly and I'll add this update at that time.

Reply:
Updated with a slightly different approach. This should help with #805 somewhat